src/ophys_etl/modules/segmentation/modules/hnc_segmentation_wrapper.py

```diff
@@ -0,0 +1,86 @@
+import argschema
+import h5py
+import json
+import networkx as nx
+from hnccorr import Movie
+from matplotlib.figure import Figure
+
+from ophys_etl.modules.segmentation.modules.schemas import \
+        HNCSegmentationWrapperInputSchema
+from ophys_etl.modules.segmentation.detect.hnc_segmentation_utils import (
+        hnc_construct, ExplicitHNCcorrCandidate, HNCcorrSegmentationObjects)
+from ophys_etl.modules.segmentation.qc_utils.roi_utils import \
+        hnc_roi_to_extract_roi
+from ophys_etl.modules.segmentation.seed.seeder import ImageMetricSeeder
+from ophys_etl.modules.segmentation.qc.seed import add_seeds_to_axes
+from ophys_etl.modules.segmentation.graph_utils.conversion import graph_to_img
+from ophys_etl.modules.segmentation.qc_utils.roi_utils import create_roi_plot
+
+
+class HNCSegmentationWrapper(argschema.ArgSchemaParser):
+    default_schema = HNCSegmentationWrapperInputSchema
+
+    def run(self):
+        self.logger.name = type(self).__name__
+
+        # define a seeder
+        seeder = ImageMetricSeeder(**self.args['seeder_args'])
+        graph_img = graph_to_img(
+                nx.read_gpickle(self.args['graph_input']),
+                attribute_name=self.args["attribute_name"])
+        seeder.select_seeds(graph_img, sigma=None)
+        # define all the HNCcorr repo objects
+        hnc_objects: HNCcorrSegmentationObjects = hnc_construct(
+                self.args["hnc_args"])
+        # read in the movie and turn it into the HNCcorr Movie object
+        with h5py.File(self.args["video_input"], "r") as f:
+            data = f["data"][()]
+        movie = Movie(name=self.args["experiment_name"], data=data)
```

Comment:
Should we do something here to make sure the raw `data` is deleted after it is converted to an HNC `Movie`? Either rename `data` -> `movie` so that the cast to `Movie` overwrites the raw data, or explicitly calling `del data` after casting to `Movie`.

(It's also possible that the amount of memory we are talking about does not actually matter)

Reply:
I believe this is passed by reference, so there is not a copy:
```
from hnccorr import Movie
data = np.zeros((100, 10, 10))
m = Movie(name="test", data=data)
print(id(data) == id(m._data))
```
```
True
```